src/neo-vm/ScriptBuilder.cs

```diff
@@ -94,14 +93,18 @@ public ScriptBuilder EmitJump(OpCode opcode, int offset)
         public ScriptBuilder EmitPush(BigInteger value)
         {
             if (value >= -1 && value <= 16) return Emit(OpCode.PUSH0 + (byte)(int)value);
-            byte[] data = value.ToByteArray(isUnsigned: false, isBigEndian: false);
-            if (data.Length == 1) return Emit(OpCode.PUSHINT8, data);
-            if (data.Length == 2) return Emit(OpCode.PUSHINT16, data);
-            if (data.Length <= 4) return Emit(OpCode.PUSHINT32, PadRight(data, 4, value.Sign < 0));
-            if (data.Length <= 8) return Emit(OpCode.PUSHINT64, PadRight(data, 8, value.Sign < 0));
-            if (data.Length <= 16) return Emit(OpCode.PUSHINT128, PadRight(data, 16, value.Sign < 0));
-            if (data.Length <= 32) return Emit(OpCode.PUSHINT256, PadRight(data, 32, value.Sign < 0));
-            throw new ArgumentOutOfRangeException(nameof(value));
+            Span<byte> buffer = stackalloc byte[32];
+            if (!value.TryWriteBytes(buffer, out int bytesWritten, isUnsigned: false, isBigEndian: false))
+                throw new ArgumentOutOfRangeException(nameof(value));
+            return bytesWritten switch
+            {
+                1 => Emit(OpCode.PUSHINT8, PadRight(buffer, bytesWritten, 1, value.Sign < 0)),
+                2 => Emit(OpCode.PUSHINT16, PadRight(buffer, bytesWritten, 2, value.Sign < 0)),
+                <= 4 => Emit(OpCode.PUSHINT32, PadRight(buffer, bytesWritten, 4, value.Sign < 0)),
+                <= 8 => Emit(OpCode.PUSHINT64, PadRight(buffer, bytesWritten, 8, value.Sign < 0)),
+                <= 16 => Emit(OpCode.PUSHINT128, PadRight(buffer, bytesWritten, 16, value.Sign < 0)),
+                _ => Emit(OpCode.PUSHINT256, PadRight(buffer, bytesWritten, 32, value.Sign < 0)),
```

Comment:
We lose `ArgumentException` if we want to push something larger than 32 bytes.

Reply:
The exception is in L98.